connector/subgraph/src/DisputableDelay.ts

```diff
@@ -0,0 +1,159 @@
+import {BigInt, Address} from '@graphprotocol/graph-ts'
+import {Agreement as AgreementContract} from '../generated/templates/Agreement/Agreement'
+import {ERC20 as ERC20Contract} from '../generated/templates/DisputableDelay/ERC20'
+import {
+  DisputableDelay as DisputableDelayEntity,
+  DelayedScript as DelayedScriptEntity,
+  ERC20 as ERC20Entity,
+} from '../generated/schema'
+import {
+  DisputableDelay as DelayContract,
+  ExecutionDelaySet as ExecutionDelaySetEvent,
+  DelayedScriptStored as DelayedScriptStoredEvent,
+  ExecutionPaused as ExecutionPausedEvent,
+  ExecutionResumed as ExecutionResumedEvent,
+  ExecutionCancelled as ExecutionCancelledEvent,
+  ExecutedScript as ExecutedScriptEvent
+} from '../generated/templates/DisputableDelay/DisputableDelay'
+
+/* eslint-disable @typescript-eslint/no-use-before-define */
+
+export function handleExecutionDelaySet(event: ExecutionDelaySetEvent): void {
+  const delayContract = DelayContract.bind(event.address)
+  const delay = loadOrCreateDelay(event.address)!
+  delay.delayedScriptsNewIndex = delayContract.delayedScriptsNewIndex()
+  delay.save()
+}
+
+export function handleDelayedScriptStored(event: DelayedScriptStoredEvent): void {
+  const disputableDelay = DisputableDelayEntity.load(event.address.toHexString())!
+
+  const delayedScriptId = buildDelayedScriptId(event.address, event.params.delayedScriptId)
+  const delayedScript = new DelayedScriptEntity(delayedScriptId)
+
+  const delayContract = DelayContract.bind(event.address)
+  const delayedScriptData = delayContract.delayedScripts(event.params.delayedScriptId)
+
+  delayedScript.disputableDelay = event.address.toHexString()
+  delayedScript.delayedScriptId = event.params.delayedScriptId
+  delayedScript.executionFromTime = delayedScriptData.value0
+  delayedScript.pausedAt = delayedScriptData.value1
+  delayedScript.delayedScriptStatus = delayedScriptStatus(delayedScriptData.value2)
+  delayedScript.evmScript = delayedScriptData.value3
+  delayedScript.actionId = delayedScriptData.value4
+  delayedScript.submitter = delayedScriptData.value5
+  delayedScript.context = event.params.context
+  delayedScript.disputeId = BigInt.fromI32(0)
+  delayedScript.challengeId = BigInt.fromI32(0)
+  delayedScript.challenger = Address.fromString('0x0000000000000000000000000000000000000000')
+  delayedScript.challengeEndDate = BigInt.fromI32(0)
+  delayedScript.settledAt = BigInt.fromI32(0)
+  delayedScript.disputedAt = BigInt.fromI32(0)
+  delayedScript.executedAt = BigInt.fromI32(0)
+  delayedScript.collateralRequirement = disputableDelay.collateralRequirement
+  delayedScript.save()
+
+  disputableDelay.delayedScriptsNewIndex = delayContract.delayedScriptsNewIndex()
```

Comment:
Not a big deal but we could prevent performing another contract call

```suggestion
  disputableDelay.delayedScriptsNewIndex = event.params.delayedScriptId.plus(BigInt.fromI32(1))
```